src/imgix-core-js.js

```diff
@@ -43,15 +44,37 @@
         this.settings[key] = val;
       }
 
-      if (!Array.isArray(this.settings.domains)) {
+      // If settings.domains is passed in as an array of 1 or more elements
+      // will generate a deprecation warning
+      if (Array.isArray(this.settings.domains) && this.settings.domains.length > 0) {
+        console.warn("Warning: Domain sharding has been deprecated and will be removed in the next major version.\nAs a result, the 'domains' argument will be deprecated in favor of 'domain' instead.");
+      }
+      // If settings.domains is passed in as a signle domain str
+      // will keep original behavior
+      else if (!Array.isArray(this.settings.domains)) {
         this.settings.domains = [this.settings.domains];
       }
+      // If new domain argument is being used
       else {
-        console.warn("Warning: Domain sharding has been deprecated and will be removed in the next major version.\nAs a result, the 'domains' argument will be deprecated in favor of 'domain' instead.");
+        // If domain is passed an array
+        // check if it is an array > 1 elements
+        if (Array.isArray(this.settings.domain)) {
```

Comment:
I don't think we should even allow `domain` to be passed as an array.

Reply:
Agreed, I figured this would be a way to mitigate it if users tried to. I'm open to other ways that accomplish the same thing.